src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -14,12 +14,67 @@ namespace py = pybind11;
 
 void regclass_Tensor(py::module m) {
     py::class_<ov::Tensor, std::shared_ptr<ov::Tensor>> cls(m, "Tensor");
+    cls.doc() = "openvino.runtime.Tensor holding either copy of memory or shared host memory.";
 
     cls.def(py::init([](py::array& array, bool shared_memory) {
                 return Common::tensor_from_numpy(array, shared_memory);
             }),
             py::arg("array"),
-            py::arg("shared_memory") = false);
+            py::arg("shared_memory") = false,
+            R"(
+                Tensor's special constructor.
+
+                Parameters
+                ----------
+                array : numpy.array
+
+                shared_memory : bool
+                    If `True` this Tensor memory is being shared with a host,
+                    that means the responsibility of keeping host memory is
+                    on the side of a user. Any action performed on the host
+                    memory will be reflected on this Tensor's memory!
+                    If `False`, data is being copied to this Tensor.
+
+                    Requires data to be C_CONTIGUOUS if `True`.
+
+                    Default: `False`
+
+                Returns
+                ----------
+                __init__ : openvino.runtime.Tensor
+            )");
+
+    cls.def(py::init([](const ov::element::Type element_type, const ov::Shape& shape, py::array& array) {
+                return Common::tensor_from_pointer(element_type, shape, array);
+            }),
+            py::arg("element_type"),
+            py::arg("shape"),
+            py::arg("array"),
+            R"(
+                Another Tensor's special constructor.
+
+                Mainly to match with C++ API, please use it only in advanced cases if necessary!
```

Comment:
I'm not sure that we should have a lot of notes that is only a C++ specific or something, because python also have such api - numpy arrays work the same - np.array(buffer, shape, dtype, copy=True). So we also can add `copy` parameter (later maybe).
Possible usage:
`tensor = Tensor(model.input().element_type, model.input().shape, data)` # make dtype conversion using openvino api instread numpy
 `request.set_input_tensor(Tensor)`